app/src/main/java/com/android/sample/ui/account/AccountScreen.kt

```diff
@@ -1,11 +1,148 @@
+
+        Column(
+            modifier = Modifier.fillMaxSize().padding(padding).padding(8.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp, Alignment.Top),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+          Spacer(modifier = Modifier.height(16.dp))
+
+          ProfilePicture(profilePictureUrl.value, modifier = Modifier.weight(.4f))
+
+          Text(
+              text = userName.value ?: stringResource(R.string.account_screen_default_user_name),
+              modifier = Modifier.weight(.1f).testTag("userName"),
+              style = MaterialTheme.typography.titleLarge)
+
+          ListSelection(userViewModel, modifier = Modifier.weight(.8f))
+        }
+      })
+}
+
+@Composable
+private fun ListSelectionButton(
+    modifier: Modifier,
+    title: String,
+    onClick: () -> Unit,
+    isSelected: Boolean = false
+) {
+  Box(
+      modifier = modifier.clickable { onClick() },
+      contentAlignment = Alignment.Center,
+  ) {
+    Text(
+        modifier = Modifier,
+        text = title,
+        style = MaterialTheme.typography.titleSmall,
+        color =
+            if (isSelected) MaterialTheme.colorScheme.onPrimary
+            else MaterialTheme.colorScheme.onTertiary)
+  }
+}
+
+@Composable
+private fun ListSelection(userViewModel: UserViewModel, modifier: Modifier = Modifier) {
+  val likedRecipes = userViewModel.likedRecipes.collectAsState()
+  val createdRecipes = userViewModel.createdRecipes.collectAsState()
+
+  var selectedList by remember { mutableStateOf(likedRecipes) }
+  var selectedListIndex by remember { mutableIntStateOf(0) }
+
+  Column(
+      modifier = modifier,
+      verticalArrangement = Arrangement.spacedBy(16.dp, Alignment.Top),
+      horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    Row(modifier = Modifier.height(30.dp).fillMaxWidth()) {
+      ListSelectionButton(
+          modifier = Modifier.weight(1f).testTag("likedRecipesButton"),
+          onClick = {
+            selectedList = likedRecipes
+            selectedListIndex = 0
+          },
+          title = stringResource(R.string.account_screen_liked_recipe_button_title),
+          isSelected = selectedListIndex == 0)
+      ListSelectionButton(
+          modifier = Modifier.weight(1f).testTag("createdRecipesButton"),
+          onClick = {
+            selectedList = createdRecipes
+            selectedListIndex = 1
+          },
+          title = stringResource(R.string.account_screen_created_recipe_button_title),
+          isSelected = selectedListIndex == 1)
+    }
+
+    HorizontalDivider(
+        modifier = Modifier.fillMaxWidth(.8f),
+        thickness = 2.dp,
+        color = MaterialTheme.colorScheme.onTertiary)
+
+    RecipeList(list = selectedList.value, modifier = Modifier.weight(1f).fillMaxWidth())
+  }
+}
+
+@Composable
+private fun ProfilePicture(profilePictureUrl: String?, modifier: Modifier = Modifier) {
```

Comment:
I noticed you're using a string for the `profilePictureUrl`. Will this be a Firebase UID, or do you have another approach in mind? One idea I had is to leverage the Google account profile picture, given that users sign in with Google. 

In a cooking app, personalization of the profile picture might not be critical, so simplifying it to this mechanism likely wouldn’t detract from the user experience. That said, this should be discussed with the whole team to ensure everyone’s on the same page. I’d also be interested in hearing your thoughts on this approach.

For reference, [here’s](https://github.com/composablehorizons/firebase-auth-tutorial/blob/main/app/src/main/java/co/composables/firebaseauth/MainActivity.kt) is a repo implementing this feature.

Reply:
I didn't know we good use the google account profile picture. We could definitely implement this later, but for now, I think this should do. Additionally, we're probably going to switch from the Url to the imageUid system to use the firestorage  db.